src/axom/sidre/core/MFEMSidreDataCollection.cpp

```diff
@@ -873,21 +883,36 @@ void MFEMSidreDataCollection::Load(const std::string& path,
   // the domain and global groups are, and can restore them after the Load().
   //
   // If the data collection did not create the datastore, the host code must
-  // reset these pointers after the load operation and also reset the state
-  // variables.
+  // reset these pointers after the load operation and also reset the state variables.
   if(m_owns_datastore)
   {
     // Use the same path format as was used to create the datastore
     SetGroupPointers(m_datastore_ptr->getRoot()->getGroup(
                        name + "_global/blueprint_index/" + name),
                      m_datastore_ptr->getRoot()->getGroup(name));
-    SLIC_ERROR_IF(m_bp_grp->getNumGroups() == 0,
-                  "Loaded datastore is empty, was the datastore created on a "
-                  "different number of nodes?");
+    SLIC_ERROR_IF(
+      m_bp_grp->getNumGroups() == 0,
+      fmt::format("Loaded datastore is empty, was the datastore created on a "
+                  "different number of ranks? Current run has {} ranks, but "
+                  "dataset was created using {} ranks",
+                  numCommRanks,
+                  numInputRanks));
 
     UpdateStateFromDS();
     UpdateMeshAndFieldsFromDS();
   }
+
+  // Set the mesh nodal grid function when present
+  const std::string gfPath = "topologies/mesh/grid_function";
+  if(GetBPGroup()->hasView(gfPath))
+  {
+    const std::string nodalGFName = GetBPGroup()->getView(gfPath)->getString();
+    if(this->HasField(nodalGFName))
+    {
+      this->SetMeshNodesName(nodalGFName);
+      mesh->NewNodes(*this->GetField(nodalGFName), false);
+    }
+  }
```

Comment:
DC bugfix 3:  We need to account for the nodal grid function when loading an `MFEMSidreDataCollection`.